### PR TITLE
add bpf-filter comments

### PR DESCRIPTION
**Pull request by dushyantbehl**

### CODE REVIEW

File: op/bpf-filter-master/commented_bpf-filter-master/drop.c [hunk 1]
```diff
@@ -126,10 +126,10 @@ struct bpf_elf_map iface_stat_map __section("maps") = {
   ],
   "humanFuncDescription": [
     {
-      "description": "",
-      "author": "",
-      "authorEmail": "",
-      "date": ""
+      "description": "This function compares mac addresses represented by two unsigned char arrays of length 6 mac1 and mac2 passed as arguments, returns 1 if true else false",
```
Comment (sdsen) [inline op/bpf-filter-master/commented_bpf-filter-master/drop.c:129]: @dushyantbehl  maybe reword to Returns 1 if they are same, 0 otherwise ?

File: op/bpf-filter-master/commented_bpf-filter-master/drop.c [hunk 2]
```diff
@@ -508,10 +508,10 @@ static __inline int is_broadcast_mac(__u8 *m) {
   ],
   "humanFuncDescription": [
     {
-      "description": "",
-      "author": "",
-      "authorEmail": "",
-      "date": ""
+      "description": "This function is a filter to be attached on the root interface of a pod's veth pair at TC layer. It takes in a packet in sk_buff form and performs a MAC and IP filter on top. It first checks if the packet is well formed, reads a map called iface_map with the key as ingress interface of the packet, it also queries the iface_ip_map with the same key. First map returns the mac address which is allowed to pass through the interface and second contains the ip address which is allowed. The filter applied is of this form, allow packets coming from the pod only in these cases, 1) if the source or dest mac are broadcast addresses then allow, 2) if the packet source mac address matches that of the pod then allow, 3) if the packet source ip matches that of the pod then allow. It also lets the packet pass if it is going towards the pod. It also prints the matching or unmatching mac or ip addresse. Returns TC_ACT_OK if filter passes else TC_ACT_SHOT. The filter also records the PASS or SHOT statistics in the map iface_stat_map",
```
Comment (sdsen) [inline op/bpf-filter-master/commented_bpf-filter-master/drop.c:511]: Can you specify if all or any of the filters need to match.

**Review by sdsen — CHANGES_REQUESTED** [review]

**Comment by dushyantbehl** [comment]
@sdsen updated...you can check again

File: op/bpf-filter-master/commented_bpf-filter-master/drop.c (hunk 2 above)
Comment (dushyantbehl) [reply op/bpf-filter-master/commented_bpf-filter-master/drop.c]: Fixed.

File: op/bpf-filter-master/commented_bpf-filter-master/drop.c (hunk 1 above)
Comment (dushyantbehl) [reply op/bpf-filter-master/commented_bpf-filter-master/drop.c]: Fixed.

**Review by sdsen — APPROVED** [review]
lgtm
